### PR TITLE
Fix/daef 464 Fix broken makePayment js-api function

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR introduces a proper way of handling wallet spending password in JS Api which must be included in the request as a part of query string and it's value must be encrypted.

### CODE REVIEW

File: app/api/js-api/lib/request.js [hunk 1]
```diff
@@ -14,20 +16,43 @@ export type RequestOptions = {
   },
 };
 
+const bytesToB16 = (bytes) => Buffer.from(bytes).toString('hex');
+const blake2b = (data) => blakejs.blake2b(data, null, 32);
+const encryptPasspharse = (passphrase) => bytesToB16(blake2b(passphrase));
```
Comment (DominikGuzei) [inline app/api/js-api/lib/request.js]: Small typo in `encryptPasspharse` ;)

Comment (nikolaglumac) [reply app/api/js-api/lib/request.js]: Oh sorry for that... Good catch ;)

Comment (nikolaglumac) [reply app/api/js-api/lib/request.js]: @DominikGuzei typos fixed ;)

**Comment by nikolaglumac** [comment]
Closing this PR in favour of: https://github.com/input-output-hk/daedalus/pull/525 (where all the fixes from this PR are actually implemented)

**Pull request closed without merging**